homeassistant/components/instructure/calendar.py

```diff
@@ -57,7 +60,7 @@ def event(self) -> CalendarEvent | None:
 
     def get_assignments_with_due_date(self):
         assignments = self.coordinator.data[ASSIGNMENTS_KEY].values()
-        return [i for i in assignments if i["due_at"]]
+        return [i for i in assignments if "due_at" in i]
```

Comment:
"due at" field always exists as long as the data is fetched from API. However the value can be null so I think the old code is correct.

I've checked postman as well. "due_at": null happens for Software Evolotion Project course as "Group Milestone 2:  Description Summary" has no due date.